src/ui/components/shared/WorkspaceSettingsModal/GeneralSettings.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from "classnames";
+import React, { useEffect, useRef, useState } from "react";
+import hooks from "ui/hooks";
+
+const Label = ({ className, children }: { className?: string; children: React.ReactNode }) => {
+  return (
+    <div className={classNames(className, "w-4/12 flex-shrink-0")}>
+      <label>{children}</label>
+    </div>
+  );
+};
+
+const Input = ({ children }: { children: React.ReactNode }) => {
+  return <div className="w-8/12">{children}</div>;
+};
+
+const Row = ({ children }: { children: React.ReactNode }) => {
+  return <div className="flex items-center">{children}</div>;
+};
+
+const GeneralSettings = ({ workspaceId }: { workspaceId: string }) => {
+  const { workspace } = hooks.useGetWorkspace(workspaceId!);
+  const updateWorkspaceSettings = hooks.useUpdateWorkspaceSettings();
+  const [name, setName] = useState(workspace?.name);
```

Comment:
Is it okay to set name to undefined? It looks like there's going to have to be some loading first here.